pkg/storegateway/gateway_test.go

```diff
@@ -295,11 +294,6 @@ func TestStoreGateway_InitialSyncWithWaitRingTokensStability(t *testing.T) {
 			bucketIndexEnabled := bucketIndexEnabled
 			t.Run(fmt.Sprintf("%s (bucket index enabled = %v)", testName, bucketIndexEnabled), func(t *testing.T) {
 				t.Parallel()
-				// Randomize the seed but log it in case we need to reproduce the test on failure.
-				seed := time.Now().UnixNano()
-				rand.Seed(seed)
-				t.Log("random generator seed:", seed)
-
 				ctx := context.Background()
```

Comment:
If this test fails, how will we reproduce the failure if we don't know how was random number generator seeded?

I would suggest to create custom rng, use it in the test and log the seed as before.

Reply:
@pstibrany the test doesn't use `rand` though. Are you thinking of possible implicit uses of `rand` in code that the test calls?

Reply:
I see. In these tests gateway do generate random tokens when joining ring. But if we wanted to pass explicit rng, it would require more changes. I agree with removal without replacement.